src/runtime_src/hip/core/event.cpp

```diff
@@ -15,25 +15,24 @@ void event::record(std::shared_ptr<stream> s)
   cstream = std::move(s);
   auto ev = std::dynamic_pointer_cast<event>(command_cache.get(static_cast<command_handle>(this)));
   throw_invalid_handle_if(!ev, "event passed is invalid");
-  /*
-  TODO
   if (is_recorded()) {
     // already recorded
-    cstream->erase_cmd(ev); It is commented will be uncomment after stream PR is pushed
+    cstream->erase_cmd(ev);
   }
   // update recorded commands list
-  cstream->enqueue_event(std::move(ev));It is commented will be uncomment after stream PR is pushed
-  */
+  cstream->enqueue_event(ev);
   set_state(state::recorded);
 }
 
 bool event::is_recorded() const
 {
+  //the event is recorded only if the state is not init
   return get_state() >= command::state::recorded;
 }
 
 bool event::query()
 {
+  //This function will return true if all commands in the appropriate stream which specified to hipEventRecord() have completed.
   for (auto rec_com : recorded_commands){
```

Comment:
May want to double check and change all range-based for loops to use `auto&` when not a fundamental type.  In this case rec_com is a shared_ptr, copying is not for free.